scripts/release.py

```diff
@@ -414,62 +502,119 @@ def build_parser():
         '--docs-repo-url', help='URL of docs repo to update.',
         default='ssh://git@github.com/rackerlabs/docs-rpc.git',
     )
+    # Workflow args
     parser.add_argument(
-        '--tag', type=Tag, required=True, help='Name of new tag.'
+        '--do-not-publish-release', default=False, action='store_true',
+        help='Do not publish a github release for this release'
     )
-    existing = parser.add_mutually_exclusive_group(required=True)
-    existing.add_argument(
-        '--commit', help='Reference to the commit to tag.'
+    parser.add_argument(
+        '--do-not-update-milestones', default=False, action='store_true',
+        help='Do not update github milestones'
     )
-    existing.add_argument(
-        '--existing-release', default=False, action='store_true'
+    parser.add_argument(
+        '--do-not-file-docs-issue', default=False, action='store_true',
+        help='Do not create a docs issue for this release'
+    )
+    parser.add_argument(
+        '--do-not-change-files-with-release-version', default=False,
+        action='store_true',
+        help='Do not update files with the new rpc_release: "future tag"'
     )
-    parser.add_argument('--github-token')
     return parser
 
 
+def validate_args(args):
+    ''' Ensure the args are following expectations.
+    '''
+    # Store tag as a string on top of as a tag object
+    args.tag_str = str(args.tag)
+    # Future_tag must be str or None
+    if args.future_tag:
+        args.future_tag_str = str(args.future_tag)
+    # args.branch is always a string, discovered or not
+    if not args.branch:
+        branches = (b.strip() for b in
+                    git("branch", "--no-color").stdout.split("\n"))
+        for branch in branches:
+            if branch.startswith('*'):
+                args.branch = branch[2:]
+                break
+        else:
+            raise SystemExit("The current branch cannot be discovered")
+    # Github token must be in CLI or env var
+    if not args.github_token:
+        token = os.environ.get('RPC_GITHUB_TOKEN', False)
+        if token:
+            args.github_token = token
+        else:
+            raise SystemExit("Token neither found in the CLI nor in env vars")
+    # RPC version should be in CLI or env var, as a str(yes|no), default yes.
+    if args.rpc_version_check:
+        chk_state = args.rpc_version_check
+    else:
+        chk_state = os.environ.get('RPC_VERSION_CHECK', 'yes')
+    args.version_check = True if chk_state == 'yes' else False
```

Comment:
Not requesting a change, just informational
```python
True if chk_state != "no" else False
```
is equivalent to
```python
chk_state != "no"
```
I guess it could be argued either way as to which is more readable given the use of a negative.